packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -867,6 +871,34 @@ export class SeedlessOnboardingController<EncryptionKey> extends BaseController<
     );
   }
 
+  /**
+   * Get the access token from the state or the vault.
+   * If the access token is not in the state, it will be retrieved from the vault by decrypting it with the password.
+   *
+   * If both the access token and the vault are not available, an error will be thrown.
+   *
+   * @param password - The password to unlock the vault.
+   * @returns The access token.
+   */
+  async getAccessToken(password: string): Promise<string> {
```

Comment:
why is this a public function? if there is no good reason for making this public, i think it should be an internal function.